Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/FindByCriteriaOperation.php

```diff
@@ -0,0 +1,170 @@
+ *
+ *  q(node).findByCriteria('Neos.NodeTypes:Text', 'text*="Neos"')
+ *
+ *  Example (node type with property filter and pagination):
+ *
+ *   q(node).findByCriteria('Neos.NodeTypes:Document', 'title*="Flow"', {limit:10, offset:2})
+ */
+class FindByCriteriaOperation extends AbstractOperation
+{
+    use CreateNodeHashTrait;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @var string
+     */
+    protected static $shortName = 'findByCriteria';
+
+    /**
+     * {@inheritdoc}
+     *
+     * @var integer
+     */
+    protected static $priority = 100;
+
+    /**
+     * @Flow\Inject
+     * @var ContentRepositoryRegistry
+     */
+    protected $contentRepositoryRegistry;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param array<int,mixed> $context (or array-like object) onto which this operation should be applied
+     * @return boolean true if the operation can be applied onto the $context, false otherwise
+     */
+    public function canEvaluate($context)
+    {
+        foreach ($context as $contextNode) {
+            if (!$contextNode instanceof Node) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+    /**
+     * This operation operates rather on the given Context object than on the given node
+     * and thus may work with the legacy node interface until subgraphs are available
+     * {@inheritdoc}
+     *
+     * @param FlowQuery<int,mixed> $flowQuery the FlowQuery object
+     * @param array<int,mixed> $arguments the arguments for this operation
+     * @throws FlowQueryException
+     * @throws \Neos\Eel\Exception
+     * @throws \Neos\Eel\FlowQuery\FizzleException
+     */
+    public function evaluate(FlowQuery $flowQuery, array $arguments): void
+    {
+        /** @var array<int,Node> $contextNodes */
+        $contextNodes = $flowQuery->getContext();
+        if (count($contextNodes) === 0) {
+            return;
+        }
+
+        $firstContextNode = reset($contextNodes);
+        assert($firstContextNode instanceof Node);
+
+        $nodeTypeFilter = $arguments[0] ?? null;
+        $propertyValueFilter = $arguments[1] ?? null;
+        $pagination =  $arguments[2] ?? null;
+
+        assert($nodeTypeFilter === null || is_string($nodeTypeFilter));
+        assert($propertyValueFilter === null || is_string($propertyValueFilter));
+        assert($pagination === null || is_array($pagination));
+
+        /** @var Node[] $result */
+        $result = [];
+        $findDescendentNodesFilter = FindDescendantNodesFilter::create(
+            nodeTypes: $nodeTypeFilter ? NodeTypeCriteria::fromFilterString($nodeTypeFilter) : null,
+            propertyValue: $propertyValueFilter ? PropertyValueCriteriaParser::parse($propertyValueFilter) : null,
+            pagination: $pagination ? Pagination::fromArray($pagination) : null
+        );
+
+        /** @var Node $contextNode */
+        foreach ($flowQuery->getContext() as $contextNode) {
```

Comment:
maybe in this "next" generation of flowquery we want to forbid the functionality that we iterate over all context nodes and either throw if there are multiple nodes or just always pick the first one. Operating on multiple nodes might not give the expected ordering as one might think

Reply:
The one thing that i dislike most is that context can contain anything. Would really like to limit that to nodes